src/components/structures/RoomView.js

```diff
@@ -1746,15 +1884,38 @@ module.exports = React.createClass({
                 hideMessagePanel = true;
             }
 
-            var messagePanel = (
+            var messagePanel;
+
+            // just show a spinner while the timeline loads.
+            //
+            // put it in a div of the right class so that the order in the
+            // roomview flexbox is correct.
+            //
+            // Note that the click-on-search-result functionality relies on the
+            // fact that the messagePanel is hidden while the timeline reloads,
+            // but that the RoomHeader (complete with search term) continues to
+            // exist.
+            if (this.state.timelineLoading) {
+                messagePanel = (
+                        <div className="mx_RoomView_messagePanel" style={{display: "flex"}}>
```

Comment:
We tend to only hard-code styles when they are for toggling visibility, rather than for modifying style-specific stuff like `flex`. Any reason why you did this here and not in the CSS, given it never changes?